### PR TITLE
Add 3keyecosystem and 2key2 keyboard

### PR DESCRIPTION
**Pull request by softplus**
Added a new group ("3keyecosystem") and a 2x1 macro-keyboard there ("2key2").

## Description

I made a modular keyboard base-board and built a 2x1 macro-keyboard for it.
This includes a default and a "via" keymap for it.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/3keyecosystem/2key2/2key2.c [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include "2key2.h"
+#include "quantum/rgb_matrix.h"
```
Comment (drashna) [inline keyboards/3keyecosystem/2key2/2key2.c:18]:
```suggestion
```

File: keyboards/3keyecosystem/2key2/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "quantum.h"
```
Comment (drashna) [inline keyboards/3keyecosystem/2key2/keymaps/default/keymap.c:16]:
```suggestion
```

File: keyboards/3keyecosystem/2key2/keymaps/via/keymap.c [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "quantum.h"
```
Comment (drashna) [inline keyboards/3keyecosystem/2key2/keymaps/via/keymap.c:16]:
```suggestion
```

File: keyboards/3keyecosystem/2key2/rules.mk [hunk 4]
```diff
@@ -0,0 +1,26 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = caterina
+
+# Build Options
+#   change yes to no to disable
+#
+DEFAULT_FOLDER = 3keyecosystem/2key2
```
Comment (drashna) [inline keyboards/3keyecosystem/2key2/rules.mk:10]: Move this entire file to the 2key2 folder, and you don't need this. 
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by softplus** [comment]
Thanks! Fixed.

File: keyboards/3keyecosystem/2key2/2key2.c [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  {   0, 1 }
+}, {
+  // LED Index to Physical Position
+  // Using range { 0..224, 0..64 }
+  { 0,  32 }, { 224,  32 }
+}, {
+  // LED Index to Flag
+  // https://docs.qmk.fm/#/feature_rgb_matrix?id=flags
+  LED_FLAG_ALL, LED_FLAG_ALL
+} };
```
Comment (drashna) [inline keyboards/3keyecosystem/2key2/2key2.c:33]:
```suggestion
#ifdef RGB_MATRIX_ENABLE
led_config_t g_led_config = { {
  // Key Matrix to LED Index
  {   0, 1 }
}, {
  // LED Index to Physical Position
  // Using range { 0..224, 0..64 }
  { 0,  32 }, { 224,  32 }
}, {
  // LED Index to Flag
  // https://docs.qmk.fm/#/feature_rgb_matrix?id=flags
  LED_FLAG_ALL, LED_FLAG_ALL
} };
#endif
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Comment by softplus** [comment]
Just checking in here to see if I need to do something, or just be more patient :-). Thanks!

File: keyboards/3keyecosystem/2key2/rules.mk [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+RGB_MATRIX_ENABLE = yes		# Enable RGB matrix
+RGB_MATRIX_DRIVER = WS2812	# Select WS2812 driver for RGB matrix
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
```
Comment (fauxpark) [inline keyboards/3keyecosystem/2key2/rules.mk:24]:
```suggestion
```

File: keyboards/3keyecosystem/2key2/rules.mk [hunk 7]
```diff
@@ -0,0 +1,25 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = caterina
+
+# Build Options
+#   change yes to no to disable
+#
+
```
Comment (fauxpark) [inline keyboards/3keyecosystem/2key2/rules.mk:10]:
```suggestion
```

File: keyboards/3keyecosystem/2key2/readme.md [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+* Hardware Supported: [3keyecosystem-2key2](https://github.com/softplus/3keyecosystem/tree/main/2key2)
+* Hardware Availability: Open-Source files at above link
+
+Make example for this keyboard (after setting up your build environment):
+
+    `make 3keyecosystem/2key2:default`
+
+or flash it (requires hitting the reset button on the main board):
+
+    `make 3keyecosystem/2key2:default:flash`
```
Comment (fauxpark) [inline keyboards/3keyecosystem/2key2/readme.md:18]:
```suggestion
    make 3keyecosystem/2key2:default

Flashing example for this keyboard:

    make 3keyecosystem/2key2:default:flash
```

File: keyboards/3keyecosystem/2key2/info.json [hunk 9]
```diff
@@ -0,0 +1,15 @@
+{
+    "keyboard_name": "3-Key-Ecosystem 2key2: 2x1 macro-keyboard",
+    "url": "https://github.com/softplus/3keyecosystem/tree/main/2key2",
+    "maintainer": "softplus",
+    "width": 2,
+    "height": 1,
```
Comment (fauxpark) [inline keyboards/3keyecosystem/2key2/info.json:6]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by softplus** [comment]
Thanks, updated!

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!n

**Comment by softplus** [comment]
Thanks for the reviews, folks!

**Pull request merged**